index.html

```diff
@@ -0,0 +1,475 @@
+
+```graphql
+#::[Using `csdl_Never` and `@csdl_error` to mark a field error]
+type SomeType {
+  someField: csdl_Never
+    @csdl_error(message: "someField defined with conflicting types: Int, String")
+}
+```
+
+## Scalars
+
+Valid CSDL documents **must** include the declarations of any CSDL scalars they use. Producers **should** include the `@specifiedBy` directive, but it is not mandatory.
+
+### `csdl_SelectionSet`
+
+```graphql
+#::[`scalar csdl_SelectionSet` declaration]
+scalar csdl_SelectionSet
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_selectionset")
+```
+
+A GraphQL selection set represented with the same syntax as defined in [the GraphQL SDL](https://spec.graphql.org/draft/#sec-Selection-Sets).
+
+Note: Unlike the federation `_FieldSet` scalar, a `csdl_SelectionSet` must parse as a GraphQL selection set, so the selections must be surrounded by braces.
+
+All legal selection set features are allowed. That includes alises, field arguments, directives, and inline fragments. Named fragment spreads are syntactically valid, but meaningless, as there is nowhere to define named fragments; future versions of this spec may change this.
+
+### `csdl_URL`
+
+```graphql
+#::[`scalar csdl_URL` declaration]
+scalar csdl_URL
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_url")
+```
+
+A [URL](https://www.w3.org/Addressing/URL/url-spec.html). Used by `@csdl_endpoint`.
+
+## Directives
+
+Valid CSDL documents **must** include the declarations of any CSDL directives they use.
+
+### `@csdl_key`
+
+```graphql
+#::[`@csdl_key` declaration]
+directive @csdl_key(graph: csdl_Graph!, repr: csdl_SelectionSet!)
+  repeatable on OBJECT
+```
+
+Define an entity key for this type within a subgraph.
+
+The `@csdl_key` directive tells consumers what subset of fields are necessary to identify this type of entity to a particular subgraph. It provides a way for csdl consumers to "switch graphs" when planning a query. For example:
+
+```graphql
+#::[A query which requires porting the `Astronaut` type between subgraphs]
+query {          # resolver                    | realizes
+                 # ----------------------------|--------------
+  rockets {      # rockets::Query.rockets     -> rockets::Rocket
+    captain {    # rockets::Rocket.astronaut  -> rockets::Astronaut
+      name       # astronauts::Astronaut.name -> String
+      rocket {   # rockets::Astronaut.rocket  -> rockets::Rocket
+        id       # rockets::Rocket.id         -> ID
+      }
+    }
+  }
+  
+}
+```
+
+`Astronaut.name` is provided by the `astronauts` subgraph. But the `captain` field is provided by the `rockets` subgraph.
+
+The fields specified in `repr` will be passed to the subgraph's `Query._entities(representations:)` as an item within the `representations` list.
+
+Multiple `@csdl_key`s can be provided for different graphs, or for the same graph.
```

Comment:
Is it worth explaining why you might have multiple for the same graph? (It's because there might be multiple sets of fields you can use to resolve a representation, right?)